emu/src/cpu/data_processing.rs

```diff
@@ -36,7 +36,7 @@ impl Arm7tdmi {
         // Shift Type (0=LSL, 1=LSR, 2=ASR, 3=ROR)
 
         let (result, carry) = match shift_type {
-            0 | 1 => {
+            0 | 1 | 2 => {
                 let a = alu_instruction::shift(
                     shift_amount.into(),
```

Comment:
Shouldn't it be `shift_type.into()` here?

Reply:
Nothing change I guess :)

Reply:
It does, no? `shift_type` is the type of the shift we want to perform. If we pass `shift_amount.into()` as a shift type we don't execute the correct shift

Reply:
Oh sorry yes! I miss read... This is the problem of doing to much TryFrom/From... Semantic error should be more explicit instead it is sneaky :(...
Fixed